lib/xdr/base.ex

```diff
@@ -44,47 +44,142 @@ defmodule XDR.Base do
   @doc false
   defmacro __before_compile__(_env) do
     quote do
+      @doc """
+      Encode an XDR value struct into its binary representation.
+      See `XDR.encode/1` for more details.
+      """
+      @spec encode(XDR.Type.t()) :: {:ok, binary()} | {:error, any()}
       defdelegate encode(type_with_value), to: XDR
+
+      @doc """
+      Like `encode/1`, but returns binary on success instead of a tuple, and raises on failure.
```

Comment:
Will mention only once, but we may opt to wrap comments at 80 columns. A VS Code Extension well-suited to this purpose (that I love) is called `Rewrap`. Enclosing details.

```
Name: Rewrap
Id: stkb.rewrap
Description: Re-wraps comments and other text to a given line length.
Version: 1.9.1
Publisher: stkb
VS Marketplace Link: https://marketplace.visualstudio.com/items?itemName=stkb.rewrap
```